### DISCUSSION TITLE
Nelua slower than luajit in this little benchmark?

### DISCUSSION THREAD

**Discussion by fenix272** [edubart/nelua-lang#263] [General]
`require 'os'
local function x()
  local j = 0
  for i=2.0, 1000000000.0 do
     j = i
  end
  return j
end

x()

local t1 = os.clock()
local y = x();
local t2 = os.clock()
print(y)

--assert(y == 1000000000.0)
print("time taken ", t2-t1);`

Results:
Nelua 4.18
Luajit  0.61
If i change the numbers to integer in the x function Nelua improves getting 1.65 whereas luajit is still about 0.61
Why is nelua slower?

**Comment by jrfondren** [comment]
With those times you're probably comparing `nelua bench.lua` vs. `luajit bench.lua`. The first step includes a lengthy compilation process in the time. Try `nelua -M -o bench bench.nelua` and then comparing the times of `./bench` and `luajit bench.lua`. With floats luajit's still 3 times faster. With integers Nelua's <strong>880</strong> times faster. I imagine that the speed advantage with floats is that luajit (due to the JIT, and due to Lua 5.1's weak integer support making this a very important optimization) notices at runtime that it doesn't need to use floating point, and uses integers instead.

**Reply by fenix272** [reply]
No. First i compiled the program and then i lauched the exe file. But either way the result is the same, 4.2

**Reply by jrfondren** [reply]
Right, you're comparing os.clock() in the running program, so you're excluding startup time. That's quite significant though:
```
$ hyperfine './bench' 'luajit bench.nelua'
Summary
  luajit bench.nelua ran
    3.00 ± 0.04 times faster than ./bench

$ hyperfine 'nelua bench.nelua' 'luajit bench.nelua'
Summary
  luajit bench.nelua ran
   12.78 ± 0.13 times faster than nelua bench.nelua
```

**Reply by asumbek** [reply]
Try with highest optimization C flags.
